modules/checklist/module.php

```diff
@@ -0,0 +1,178 @@
+		return Plugin::$instance->experiments->is_feature_active( self::EXPERIMENT_ID );
+	}
+
+	/**
+	 * Gets user's progress from db
+	 *
+	 * @return array {
+	 *      @type bool $is_hidden
+	 *      @type int $last_opened_timestamp
+	 *      @type array $steps {
+	 *          @type string $step_id => {
+	 *              @type bool $is_marked_done
+	 *              @type bool $is_completed
+	 *          }
+	 *      }
+	 *  }
+	 */
+	public function get_user_progress_from_db() {
+		return json_decode( get_option( self::DB_OPTION_KEY ), true );
+	}
+
+	/**
+	 * Using the step's ID, get the progress of the step should it exist
+	 *
+	 * @param $step_id
+	 *
+	 * @return null|array {
+	 *      @type bool $is_marked_done
+	 *      @type bool $is_completed
+	 *  }
+	 */
+	public function get_step_progress( $step_id ) {
+		foreach ( $this->user_progress['steps'] as $id => $step ) {
+			if ( $id === $step_id ) {
+				return $this->user_progress['steps'][ $step_id ] ?? null;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Update the progress of a step
+	 *
+	 * @param $step_id
+	 * @param $step_progress
+	 * @param bool $should_update_db
+	 *
+	 * @return void
+	 */
+	public function set_step_progress( $step_id, $step_progress ) {
+		$this->user_progress['steps'][ $step_id ] = $step_progress;
+		$this->update_user_progress_in_db();
+	}
+
+	/**
+	 * @return Steps_Manager
+	 */
+	public function get_steps_manager() {
+		return $this->steps_manager;
+	}
+
+	/**
+	 * @return Wordpress_Adapter
+	 */
+	public function get_wordpress_adapter() {
+		return $this->wordpress_adapter;
+	}
+
+	public function enqueue_editor_scripts() {
+		add_action( 'elementor/editor/before_enqueue_scripts', function () {
+			$min_suffix = Utils::is_script_debug() ? '' : '.min';
+
+			wp_enqueue_script(
+				$this->get_name(),
+				ELEMENTOR_ASSETS_URL . 'js/checklist' . $min_suffix . '.js',
+				[
+					'react',
+					'react-dom',
+					'elementor-common',
+					'elementor-v2-ui',
+				],
+				ELEMENTOR_VERSION,
+				true
+			);
+
+			wp_set_script_translations( $this->get_name(), 'elementor' );
+		} );
+	}
+
+	private function setup() {
+		$this->init_user_progress();
+		$this->user_progress = $this->user_progress ?? $this->get_user_progress_from_db();
+		$this->steps_manager = new Steps_Manager( $this );
+		$this->enqueue_editor_scripts();
+	}
+
+	private function register_experiment() {
+		Plugin::$instance->experiments->add_feature( [
+			'name' => self::EXPERIMENT_ID,
+			'title' => esc_html__( 'Launchpad Checklist', 'elementor' ),
+			'description' => esc_html__( 'Launchpad Checklist feature to boost productivity and deliver your site faster', 'elementor' ),
+			'release_status' => Manager::RELEASE_STATUS_ALPHA,
+			'hidden' => true,
+		] );
+	}
+
+	private function init_user_progress() {
+		$default_settings = [
+			'is_hidden' => false,
+			'last_opened_timestamp' => time(),
+			'steps' => [],
+		];
+
+		add_option( self::DB_OPTION_KEY, wp_json_encode( $default_settings ) );
```

Comment:
looks like it will re-write options with default settings on each constructor call. Or maybe I wrong?

Reply:
add_option doesn't update existing options, I can leave a comment with reference link if you think it's needed